pkg/node/manager/interfaces.go

```diff
@@ -0,0 +1,12 @@
+//go:generate mockgen --source interfaces.go --destination mocks.go --package manager
+package manager
+
+import (
+	"context"
+
+	"github.com/bacalhau-project/bacalhau/pkg/models"
+)
+
+type NodeEventHandler interface {
+	HandleNodeEvent(ctx context.Context, info models.NodeInfo, event NodeEvent)
+}
```

Comment:
I would find the code slightly more readable if we declared this interface where it's expected/used before the `RegisterHandler` method on the `NodeEventEmitter` instead of in a separate file.

Reply:
Reasonble, done in ff7828f3f.  I think I put it in an interfaces.go as there's been a bit of a pattern with the eval stuff that there's an interfaces.go and a types.go.